### DISCUSSION TITLE
not persistent cookie values after enter in another route

### DISCUSSION THREAD

**Discussion by lokize** [expressjs/session#1043] [Q&A] [answered]
hi, in my application i seted a values to cookie params, like img: 
![image](https://user-images.githubusercontent.com/19701630/197306676-faa156fb-7a4b-474c-b565-f788d0b90019.png)

![image](https://user-images.githubusercontent.com/19701630/197308565-6303fd06-88f7-4f00-8545-3a8a0aa44272.png)

but when i change to another route of application when i try to get session the values of cookies params is empty...

![image](https://user-images.githubusercontent.com/19701630/197308558-7f069e8c-456b-49c8-afcf-46bd667e8bb2.png)

how to save session in memory to prevent to lose values?

**Comment by mirker21** [comment]
I am experiencing the same problem, am hoping to find a solution.

**Comment by kkmr2011** [comment]
Having same problem

**Comment by joewagner** [comment, marked as answer]
HI @lokize @kkmr2011 and @mirker21 
Looks like you might not be using the `session.cookie` property correctly.  The valid settable properties are listed in the [readme](https://github.com/expressjs/session?tab=readme-ov-file#reqsessioncookie).
They are `cookie.expires` and `cookie.maxAge`, you can't set `role` or `client` and if you did it would not have any effect.
If you are hoping to save those values in the session, you might want to try doing something like
`req.session.userClient = response.data.user.profile.client`
